### PR TITLE
Update Doctrine UTF8 docs

### PR DESCRIPTION
**Pull request by mcfedr**
Since its now possible (https://github.com/doctrine/DoctrineBundle/pull/420) to set the default character set for all tables using DoctrineBundle it can be mentioned in the docs

### CODE REVIEW

File: book/doctrine.rst [hunk 1]
```diff
@@ -144,6 +140,60 @@ for you:
...
+
+                <doctrine:config>
+                    <doctrine:dbal
+                        driver="%database_driver%"
+                        host="%database_host%"
+                        dbname="%database_name%"
+                        user="%database_user%"
+                        password="%database_password%">
+                            <doctrine:default-table-option name="charset">utf8</doctrine:default-table-option>
+                            <doctrine:default-table-option name="collate">utf8_unicode_ci</doctrine:default-table-option>
```
Comment (xabbuh) [inline book/doctrine.rst]: We should use `utf8mb4` and `utf8mb4_unicode_ci` here too.

Comment (xabbuh) [inline book/doctrine.rst]: and we can omit the other options here to be consistent with the YAML example (same below in the PHP config)

File: book/doctrine.rst [hunk 2]
```diff
@@ -145,6 +141,54 @@ for you:
...
+                    </doctrine:dbal>
+                </doctrine:config>
+            </container>
+
+        .. code-block:: php
+
+            // app/config/config.php
+            $configuration->loadFromExtension('doctrine', array(
+                'dbal' => array(
+                    'charset' => 'utf8mb',
```
Comment (mcfedr) [inline book/doctrine.rst]: Another typo

**Comment by mcfedr** [comment]
@xabbuh I think this pull request is no longer in progress?

File: book/doctrine.rst [hunk 3]
```diff
@@ -145,6 +141,54 @@ for you:
...
         character-set-server = utf8mb4            # Replaces utf8
 
+    You can also change the defaults for Doctrine so that the generated SQL
+    uses the correct character set.
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            doctrine:
```
Comment (xabbuh) [inline book/doctrine.rst]: Please add the filename to the top of the code block:

```rst
# app/config/config.yml
```

**Comment by xabbuh** [comment]
:+1:

File: book/doctrine.rst [hunk 4]
```diff
@@ -145,6 +141,54 @@ for you:
         collation-server     = utf8mb4_general_ci # Replaces utf8_general_ci
         character-set-server = utf8mb4            # Replaces utf8
```
Comment (weaverryan) [inline book/doctrine.rst]: could/should we also just remove this section (134-142) now?

Comment (xabbuh) [inline book/doctrine.rst]: Shouldn't we keep this so that the MySQL settings are the same accessing the database using other clients?

Comment (mcfedr) [inline book/doctrine.rst]: I think we should be changing our recommendation to use `utf8mb4` in all circumstances, just like we used to recommend `utf8` - its actually safer/better to use `utf8mb4` as it supports all utf8 chars.
With that in mind, I agree with @xabbuh that the purpose of this section is to help the user have the correct character set in all cases.

**Comment by weaverryan** [comment]
Thanks Fred! I included this in the 3.0 branch (as your PR specified), since that was the first Symfony SE version that explicitly shipped with DoctrineBundle 1.6 (which has this feature). cheers!

**Pull request merged**